File: LittleProjects/Java/SpringHibernateCRUD/src/main/java/core/service/PublisherServiceImpl.java
package core.service;

import core.model.Publisher;
import core.repository.PublisherRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.annotation.Transactional;

@org.springframework.stereotype.Service
@Transactional
public class PublisherServiceImpl implements PublisherService {
    private final PublisherRepository repository;

    @Autowired
    public PublisherServiceImpl(PublisherRepository repository) {
        this.repository = repository;
    }

    public Publisher select(Long aLong) {
        return repository.select(aLong);
    }

    public Iterable<Publisher> select() {
        return repository.select();
    }

    public Publisher insert(Publisher entity) {
        return repository.insert(entity);
    }

    public Iterable<Publisher> insert(Iterable<Publisher> entities) {
        return repository.insert(entities);
    }

    public Publisher update(Publisher entity) {
        return repository.update(entity);
    }

    public Iterable<Publisher> update(Iterable<Publisher> entities) {
        return repository.update(entities);
    }

    @Override
    public void deleteById(Long aLong) {
        repository.deleteById(aLong);
    }

    @Override
    public void deleteById(Iterable<Long> longs) {
        repository.deleteById(longs);
    }

    @Override
    public void delete(Publisher entity) {
        repository.delete(entity);
    }

    @Override
    public void delete(Iterable<Publisher> entities) {
        repository.delete(entities);
    }
}

File: LittleProjects/Scala/ScalaSpringSimpleCRUD/src/main/java/com/github/gr1f0n6x/service/PublisherServiceImpl.scala
package com.github.gr1f0n6x.service

import com.github.gr1f0n6x.model.Publisher
import com.github.gr1f0n6x.repository.PublisherRepository
import org.springframework.beans.factory.annotation.Autowired
import com.github.gr1f0n6x.model.JsonWrapper._
import spray.json._

@org.springframework.stereotype.Service
class PublisherServiceImpl extends PublisherService {
  @Autowired
  var repository: PublisherRepository = _

  override def select(id: Long) = repository.select(id).toJson.toString

  override def select() = repository.select().toJson.toString

  override def insert(entity: String) = repository.insert(entity.parseJson.convertTo[Publisher]).toJson.toString

  override def delete(id: Long) = repository.delete(id).toJson.toString

  override def update(entity: String) = repository.update(entity.parseJson.convertTo[Publisher]).toJson.toString
}
